openedxstats/apps/sites/templates/sites/sites_list.html

```diff
@@ -16,17 +16,58 @@
+
+            //    renderActionButtons
+            //    Accepts arguments of the site ID and the website's active end date to determine which buttons to render inside the row under column 'Actions'
+            //    At minimum will render the button for 'Detail' and 'Delete', 'Update' will be rendered if website is still active.
+            function renderActionButtons(siteId, activeEndDate) {
+                const detail_site = `<td class="action-button-cell"><a href="/sites/site_detail/${siteId}"><button type="button" class="btn btn-primary" aria-label="Detail" data-toggle="tooltip" data-placement="bottom" title="Detail"><span class="glyphicon glyphicon-list" aria-hidden="true"></span></button></a></td>`
+                const update_site = `<td class="action-button-cell"><a href="/sites/update_site/${siteId}"><button type="button" class="btn btn-primary" aria-label="Edit" data-toggle="tooltip" data-placement="bottom" title="Edit"><span class="glyphicon glyphicon-pencil" aria-hidden="true"></span></button></a></td>`
+                const delete_site = `<td class="action-button-cell"><a href="/sites/delete_site/${siteId}"><button type="button" class="btn btn-danger" aria-label="Delete" data-toggle="tooltip" data-placement="bottom" title="Delete"><span class="glyphicon glyphicon-trash"></span></button></a></td>`
+                return `<td style="width: 1%; vertical-align: middle;"><table><tr>${detail_site + (!activeEndDate ? update_site : '') + delete_site}</tr></table></td>`
+            } 
+            
+            //    Datatables configuration
+            //    Data is retrieved the API end point served from the view 'sites_list_json' (see views.py)
             var sitesTable = $('#sites').DataTable({
-                "lengthMenu": [[25, 50, 100, 500, -1], [25, 50, 100, 500, "All"]],
-                "order": [[5, "asc"], [4, "desc"]]
+                "processing": true,                
+                "ajax":{
+                    "url":"{% url 'sites:sites_list_json' %}", // Communicating to API endpoint serving JSON data from database
+                    "dataSrc": function(json) {                // Formatting data to be compatible with DataTables
+                        const jsonSites = {};
+                        const data = JSON.parse(json.sites);
+                        Object.keys(data).forEach(function(siteID) {
+                            const site = data[siteID];
+                            // Rendering of elements for columns 'Current Version?' & 'Actions'
+                            site.fields['isCurrentVersion'] = isCurrentVersion(site.fields.active_end_date);
+                            site.fields['renderActionButtons'] = renderActionButtons(site.pk, site.fields.active_end_date);
+                        })
+                        jsonSites['data'] = data;
+                        return jsonSites.data;
```

Comment:
I don't understand why `jsonSites` is here? We don't seem to use it, we push data into it, then return that from it.  Looks to me like we can get rid of `jsonSites`, and just `return data;` at the end.